src/styles/blocks/product.scss

```diff
@@ -0,0 +1,111 @@
+.product {
+  display: flex;
+  font-size: 16px;
+  flex-direction: column;
+  width: 200px;
+  padding: 16px;
+  border: 1px solid #f3f3f3;
+  border-radius: 5px;
+
+  &:not(:nth-child(4n)) {
+    margin-right: 51px;
+  }
+
+  &:not(:nth-last-child(-n+4)) {
+    margin-bottom: 50px;
+  }
+
+  &:nth-last-child(-n + 4) {
+    margin-bottom: 22px;
+  }
+
+  &__image {
+    padding: 16px 0;
+    margin: 0 auto;
+
+    .image {
+      width: 160px;
+      height: 134px;
+      object-fit: fill;
+    }
+  }
+
+  .product-name {
+    padding: 18px 0 4px;
+
+    &__link {
+      font-size: 12px;
+      line-height: 18px;
+      font-weight: 500;
+      color: #{$main-color};
+      text-decoration: none;
+    }
+  }
+
+  .product-code {
+    margin: 0;
```

Comment:
you can just reset default margin for `<h2>` and `<p>` tags instead of setting this rule to an each element